### PR TITLE
Add support for DeploymentSettings dependency CacheOptions

### PR DESCRIPTION
**Pull request by nyobe**
Adds support for enabling dependency caching in deployment settings.

### CODE REVIEW

File: sdk/dotnet/DeploymentSettings.cs
Comment (IaroslavTitov) [inline sdk/dotnet/DeploymentSettings.cs:1]: I think you missed a sdk-generated file, that's why wf is complaining. Should be as easy as `make build` `git add sdk` and re-push

**Review by IaroslavTitov — COMMENTED** [review]
Overall looks good, would you please add CacheOptions into DeploymentSettings examples/integ tests?

File: sdk/dotnet/DeploymentSettings.cs
Comment (IaroslavTitov) [reply sdk/dotnet/DeploymentSettings.cs:1]: Oh, also need an entry in CHANGELOG_PENDING

File: provider/cmd/pulumi-resource-pulumiservice/schema.json [hunk 1]
```diff
@@ -554,6 +554,17 @@
         }
       ]
     },
+    "pulumiservice:index:DeploymentSettingsCacheOptions": {
+      "description": "Dependency cache settings for the deployment",
+      "properties": {
+        "enabled": {
```
Comment (IaroslavTitov) [inline provider/cmd/pulumi-resource-pulumiservice/schema.json:560]: Thanks for catching this in the docs PR!
Should be `enable`, without the d, to match the API

(I personally also prefer `enabled` but consistency is more important)

**Review by IaroslavTitov — APPROVED** [review]
Should change `enabled` -> `enable`, other than that LGTM

File: provider/cmd/pulumi-resource-pulumiservice/schema.json (hunk 1 above)
Comment (nyobe) [reply provider/cmd/pulumi-resource-pulumiservice/schema.json]: updated! https://github.com/pulumi/pulumi-pulumiservice/pull/436/commits/d019a21572b11d03d52c751545f386568f9bf07a

**Pull request merged**